ex1/test_x_mix_drix.py

```diff
@@ -23,30 +19,39 @@ def test_check_win_column(self):
         ])
         assert board.is_winning(X)
 
-    def test_check_win_diagonal(self):
-        board = XOBoard([
-            [XOBoard.EMPTY, XOBoard.EMPTY, X],
-            [XOBoard.EMPTY, X, XOBoard.EMPTY],
-            [X, XOBoard.EMPTY, XOBoard.EMPTY],
-        ])
+    def test_check_win_diagonal1(self):
+        board = XOBoard([[X, PlayerToken.EMPTY, O], [PlayerToken.EMPTY, X, PlayerToken.EMPTY], [O, PlayerToken.EMPTY, X]])
+        assert board.is_winning(X)
+        board = XOBoard([[X, PlayerToken.EMPTY, O], [PlayerToken.EMPTY, X, PlayerToken.EMPTY], [O, PlayerToken.EMPTY, O]])
+        assert not board.is_winning(X)
+
+    def test_check_win_diagonal2(self):
+        board = XOBoard(
+            [[XOBoard.EMPTY, XOBoard.EMPTY, X], [XOBoard.EMPTY, X, XOBoard.EMPTY], [X, XOBoard.EMPTY, XOBoard.EMPTY], ])
         assert board.is_winning(X)
 
     def test_check_not_win(self):
-        board = XOBoard([
-            [X, X, XOBoard.EMPTY],
-            [XOBoard.EMPTY, X, X],
-            [X, X, XOBoard.EMPTY],
-        ])
+        board = XOBoard([[X, X, XOBoard.EMPTY], [XOBoard.EMPTY, X, X], [X, XOBoard.EMPTY, XOBoard.EMPTY], ])
         assert not board.is_winning(X)
 
-    def test_print(self):
+    def test_print_board1(self):
         board = XOBoard([
             [X, X, X],
             [O, O, O],
             [XOBoard.EMPTY, XOBoard.EMPTY, XOBoard.EMPTY],
         ])
-        assert 'X X X' in str(board)  # Calls board.__str__()
-        assert 'O O O' in str(board)
+        assert 'X|X|X' in str(board)  # Calls board.__str__()
+        assert 'O|O|O' in str(board)
+
+    def test_print_board2(self):
+        board = XOBoard([[X, PlayerToken.EMPTY, O],
+                         [PlayerToken.EMPTY, O, X],
+                         [O, O, X]
+                         ])
+        assert str(board).count("X") == 3
+        assert str(board).count("O") == 4
+        assert str(board)[1] == str(board)[3] == str(board)[7] == str(board)[9] == str(board)[13] == str(board)[15]
+        assert str(board)[5] == str(board)[5] == str(board)[11]
```

Comment:
This test assumes a lot about the printing.
Therefore, in case of modification to the software, It is more harmful  than helpful.
Remove these 2 lines please.